api/swaggerclient.go

```diff
@@ -0,0 +1,84 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	openapihttptransport "github.com/go-openapi/runtime/client"
+	"github.com/go-openapi/strfmt"
+	"github.com/puppetlabs/puppetdb-cli/api/client"
+	"github.com/puppetlabs/puppetdb-cli/log"
+	"github.com/puppetlabs/puppetdb-cli/log/loglevel"
+)
+
+//SwaggerClient represents a puppec-code swagger client
+type SwaggerClient struct {
+	cacert, cert, key, url, token string
+}
+
+//NewClient creates a new SwaggerClient
+func NewClient(cacert, cert, key, url, tokenFile string) Client {
+	sc := SwaggerClient{
+		cacert: cacert,
+		cert:   cert,
+		key:    key,
+		url:    url,
+		token:  tokenFile,
+	}
+	return &sc
+}
+
+//ArgError represents an argument error
+type ArgError struct {
+	msg string
+}
+
+func (e *ArgError) Error() string {
+	return e.msg
+}
+
+//GetClient configures and creates a swagger generated client
+func (sc *SwaggerClient) GetClient() (*client.PuppetdbCli, error) {
+	url, err := url.Parse(sc.url)
+	if err != nil {
+		return nil, err
+	}
+	if url.Scheme != "https" && url.Scheme != "http" {
```

Comment:
Could have a method supportedScheme on SwaggerClient that returns true if scheme is supported and this code would transform to:
```suggestion
	if ! supportedScheme(url.Scheme) {
```